mongo/integration/unified/client_entity.go

```diff
@@ -160,20 +160,36 @@ func (c *clientEntity) stopListeningForEvents() {
 	c.setRecordEvents(false)
 }
 
-func (c *clientEntity) isIgnoredEvent(event *event.CommandStartedEvent) bool {
+func (c *clientEntity) isIgnoredStartedEvent(event *event.CommandStartedEvent) bool {
 	// Check if command is in ignoredCommands.
 	if _, ok := c.ignoredCommands[event.CommandName]; ok {
 		return true
 	}
 
 	if event.CommandName == "hello" || strings.ToLower(event.CommandName) == internal.LegacyHelloLowercase {
-		_, err := event.Command.LookupErr("speculativeAuthenticate")
-		speculativeAuth := err == nil
+		// If observeSensitiveCommands is false (or unset) and hello command has been
+		// redacted at operation level, hello command should be ignored as it contained
+		// speculativeAuthenticate.
+		if (c.observeSensitiveCommands == nil || !*c.observeSensitiveCommands) &&
+			len(event.Command) == 0 {
```

Comment:
nit: I find this hard to read because the indentation makes it seem like this is the first statement in the conditional rather than part of the condition itself. If including it in the previous line makes the line too long, I'd recommend splitting it up into separate variables:

```
sensitiveCommandsIgnored := ...
redacted := len(event.Command) == 0
if sensitiveCommandsIgnored && redacted {
    ...
}
```

Reply:
Split the conditional into separate variables for readability.